components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -79,80 +81,81 @@ void MAMAci::set_grids(
 
   // Temperature[K] at midpoints
-  add_field<Required>("T_mid", scalar3d_layout_mid, K, grid_name);
+  add_field<Required>("T_mid", scalar3d_mid, K, grid_name);
 
   // Vertical pressure velocity [Pa/s] at midpoints
-  add_field<Required>("omega", scalar3d_layout_mid, Pa / s, grid_name);
+  add_field<Required>("omega", scalar3d_mid, Pa / s, grid_name);
 
   // Total pressure [Pa] at midpoints
-  add_field<Required>("p_mid", scalar3d_layout_mid, Pa, grid_name);
+  add_field<Required>("p_mid", scalar3d_mid, Pa, grid_name);
 
   // Total pressure [Pa] at interfaces
-  add_field<Required>("p_int", scalar3d_layout_int, Pa, grid_name);
+  add_field<Required>("p_int", scalar3d_int, Pa, grid_name);
 
   // Layer thickness(pdel) [Pa] at midpoints
-  add_field<Required>("pseudo_density", scalar3d_layout_mid, Pa, grid_name);
+  add_field<Required>("pseudo_density", scalar3d_mid, Pa, grid_name);
 
   // planetary boundary layer height
   add_field<Required>("pbl_height", scalar2d_layout_col, m, grid_name);
 
   // cloud fraction [nondimensional] computed by eamxx_cld_fraction_process
-  add_field<Required>("cldfrac_tot", scalar3d_layout_mid, nondim, grid_name);
+  add_field<Required>("cldfrac_tot", scalar3d_mid, nondim, grid_name);
 
-  auto m2 = pow(m, 2);
-  auto s2 = pow(s, 2);
+  constexpr auto m2 = pow(m, 2);
+  constexpr auto s2 = pow(s, 2);
 
   // NOTE: w_variance im microp_aero.F90 in EAM is at "itim_old" dynamics time
   // step. Since, we are using SE dycore, itim_old is 1 which is equivalent to
   // the current time step. For other dycores (such as EUL), it may be different
   // and we might need to revisit this.
 
   // Vertical velocity variance at midpoints
-  add_field<Required>("w_variance", scalar3d_layout_mid, m2 / s2, grid_name);
+  add_field<Required>("w_variance", scalar3d_mid, m2 / s2, grid_name);
 
   // NOTE: "cldfrac_liq" is updated in SHOC. "cldfrac_liq" in C++ code is
   // equivalent to "alst" in the shoc_intr.F90. In the C++ code, it is used as
   // "shoc_cldfrac" and in the F90 code it is called "cloud_frac"
 
   // Liquid stratiform cloud fraction at midpoints
-  add_field<Required>("cldfrac_liq", scalar3d_layout_mid, nondim, grid_name);
+  add_field<Required>("cldfrac_liq", scalar3d_mid, nondim, grid_name);
 
   // Previous value of liquid stratiform cloud fraction at midpoints
-  add_field<Required>("cldfrac_liq_prev", scalar3d_layout_mid, nondim,
-                      grid_name);
+  add_field<Required>("cldfrac_liq_prev", scalar3d_mid, nondim, grid_name);
 
   // Eddy diffusivity for heat
-  add_field<Required>("eddy_diff_heat", scalar3d_layout_mid, m2 / s, grid_name);
+  add_field<Required>("eddy_diff_heat", scalar3d_mid, m2 / s, grid_name);
 
-  // Layout for 4D (2d horiz X 1d vertical x number of modes) variables
-  const int num_aero_modes = mam_coupling::num_aero_modes();
-  FieldLayout scalar4d_layout_mid =
-      make_layout({ncol_, num_aero_modes, nlev_}, {"COL", "NMODES", "LEV"});
```

Comment:
This is done. I've noticed additional warnings for the srf_emission interface. To avoid enlarging this PR, we'll address these warnings in a subsequent, smaller PR.